lib/Dialect/TritonGPU/Transforms/Prefetch.cpp

```diff
@@ -329,6 +339,11 @@ scf::ForOp Prefetcher::createNewForOp() {
         prevDot = newOp;
         kOff += kShape;
         kRem -= kShape;
+        if (kRem == 0) {
+          // We want to delay issuing the last dot as long as possible, ideally
+          // until after the prefetch.
```

Comment:
Maybe add: Move the insertion point to before the last decomposed dot so later instructions will be added prior to it.

Reply:
I guess if there are more than two dots, the intermediate dots may still not be overlapped well?

Reply:
@manman-ren thanks, I attempted to clarify the comments to this effect!

Reply:
> I guess if there are more than two dots, the intermediate dots may still not be overlapped well? Maybe we want to handle that separately.

If the original dot is decomposed into 3 small dots, we will have
prefetch 2nd dot operands
run 1st dot
prefetch 3rd dot operands
run 2nd dot
...
run 3rd dot
...
prefetch 1st dot operands
This is more like a scheduling problem now. May make sense to do it in the scheduler :]